### PR TITLE
Go mod migration

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

- Initialize the go.mod from the existing Gopkg.lock
- Update to a more go module friendly version of [docker/docker](https://github.com/moby/moby/releases/tag/v19.03.12) using 
   ```sh
   go get github.com/docker/docker@v19.03.12
   ```
This is the latest version and should work for us based on https://github.com/moby/moby/issues/39302#issuecomment-639687466
- Update the makefile and dockerfiles to set the GOFLAGS so that the vendor folder is respected

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Closes #757

## How Has This Been Tested?

1. Installed locally using `make local-install`, tested various commands with a local cluster 
2. ensured that both `make build` and `make build redist` correctly built binaries

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module github.com/openfaas/faas-cli
+
+go 1.14
+
+// replace github.com/docker/docker => github.com/docker/engine v1.4.2-0.20190717161051-705d9623b7c1
```
Comment (alexellis) [inline go.mod:5]: This is commented out, should it be?

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Thank you. Here's the release -> https://github.com/openfaas/faas-cli/releases/tag/0.12.10

**Comment by alexellis** [comment]
@LucasRoesler PTAL? https://travis-ci.com/github/openfaas/faas-cli/builds/184081689

**Comment by LucasRoesler** [comment]
> @LucasRoesler PTAL? https://travis-ci.com/github/openfaas/faas-cli/builds/184081689

That is an odd error, i have run the redist before I made the PR 

The error says 

```sh
build github.com/openfaas/faas-cli: cannot load github.com/sirupsen/logrus: open /go/src/github.com/openfaas/faas-cli/vendor/github.com/sirupsen/logrus: no such file or directory
```

but that file is in the repo, but it looks like the committed naming is the old `Sirupsen`  instead of the correct `sirupsen` It seems like a case sensitive file system issue. 

image

I can't commit that change from my mac

**Comment by alexellis** [comment]
Can you run an Ubuntu container and do it there?

**Pull request merged**